### PR TITLE
Kill 1.8

### LINKED ISSUE #375: It's time for 1.8/1.9 support in Parser to die
- [x] Remove all 1.8-specific feature checks
 - [x] Remove all 1.9-specific feature checks because it makes no sense to keep that in either
 - [x] Update README to say "runs on all versions of Ruby since 2.0, and historically, all versions of Ruby since 1.8"

### PR DESCRIPTION
**Pull request by alexdowad**
Fixes #375

### CODE REVIEW

**Comment by bbatsov** [comment]
👍 I think these cleanup was long overdue.

File: lib/parser/ruby19.y [hunk 1]
```diff
@@ -2147,8 +2147,6 @@ end
 
 require 'parser'
 
-Parser.check_for_encoding_support
```
Comment (bbatsov) [inline lib/parser/ruby19.y]: Why didn't you delete the entire file?

**Comment by bbatsov** [comment]
Btw, MacRuby is a dead project, that was only 1.9 compatible, so I guess killing support for it also makes sense.

**Comment by alexdowad** [comment]
@bbatsov It's not clear whether @whitequark wants to kill support for
parsing Ruby 1.8 code, or for running the parser on a 1.8 VM. I assume it's
the latter.

File: lib/gauntlet_parser.rb [hunk 2]
```diff
@@ -69,13 +67,7 @@ def parse(name)
...
 
       try(Parser::Ruby20, RUBY20, file) do
-        puts "Trying 1.9:"
-        try(Parser::Ruby19, RUBY19, file, show_ok: true) do
-          puts "Trying 1.8:"
-          try(Parser::Ruby18, RUBY18, file, show_ok: true) do
-            puts "Invalid syntax."
-          end
-        end
+        puts "Invalid syntax."
```
Comment (whitequark) [inline lib/gauntlet_parser.rb]: This should actually stay in so we don't barf on 1.8-only gems.

File: lib/parser/current.rb [hunk 3]
```diff
@@ -9,24 +9,6 @@ def warn_syntax_deviation(feature, version)
   end
 
   case RUBY_VERSION
-  when /^1\.8\./
```
Comment (whitequark) [inline lib/parser/current.rb]: This should be kept and raise an error.

File: lib/parser/current.rb [hunk 4]
```diff
@@ -9,24 +9,6 @@ def warn_syntax_deviation(feature, version)
...
-    current_version = '1.8.7'
-    if RUBY_VERSION != current_version
-      warn_syntax_deviation 'parser/ruby18', current_version
-    end
-
-    require 'parser/ruby18'
-    CurrentRuby = Ruby18
-
-  when /^1\.9\./
```
Comment (whitequark) [inline lib/parser/current.rb]: Same here.

File: lib/parser/current.rb (hunk 3 above)
Comment (alexdowad) [reply lib/parser/current.rb]: Wouldn't it be better to put something in `lib/parser.rb` which either raises an error or prints a warning if run on 1.8/1.9?

If we put it in `lib/parser/current.rb`, the error will only be seen if that file is loaded.

Comment (whitequark) [reply lib/parser/current.rb]: Sure, that's better.

**Comment by bbatsov** [comment]
Seems you're right. I see little point in keeping the parsing support around, though, as there are almost no 1.8-compatible projects around these days and just a handful of 1.9-compatible ones. Dropping support for those encourages laggards to upgrade IMO.

**Comment by whitequark** [comment]
> Dropping support for those encourages laggards to upgrade IMO.

One person's old junk is another person's history. I'm not in the business of making value judgements over here.

**Comment by alexdowad** [comment]
Updated as per review from @whitequark. Let me know if version check in `parser.rb` looks ugly. Now just rebasing on the new `master`.

**Pull request merged**